R/plotLearnerPredictionPlotly.R

```diff
@@ -0,0 +1,320 @@
+    stopf("Classification: currently only 2 or 3 features plots supported in plotLearnerPredictionPlotly(), not: %i", taskdim)
+  if (td$type == "regr" && taskdim != 2L)
+    stopf("Regression: currently only 2 features plots supported in plotLearnerPredictionPlotly(), not: %i", taskdim)
+
+  measures = checkMeasures(measures, task)
+  cv = asCount(cv)
+
+  if (missing(gridsize)) {
+    gridsize = ifelse(taskdim == 1L, 500, 100)
+  } else {
+    gridsize = asCount(gridsize)
+  }
+  assertNumber(pointsize, lower = 0)
+  assertChoice(err.mark, choices = c("train", "cv", "none"))
+
+  if (td$type == "classif" && err.mark == "cv" && cv == 0L)
+    stopf("Classification: CV must be switched on, with 'cv' > 0, for err.type = 'cv'!")
+
+  # subset to features, set hyperpars
+  task = subsetTask(task, features = features)
+  learner = setHyperPars(learner, ...)
+
+  # some shortcut names
+  target = td$target
+  data = getTaskData(task)
+  y = getTaskTargets(task)
+  x1n = features[1L]
+  x1 = data[, x1n]
+
+
+  # predictions
+  # if learner supports prob or se, enable it
+  if (td$type == "regr" && taskdim == 1L && hasLearnerProperties(learner, "se"))
+    learner = setPredictType(learner, "se")
+  if (td$type == "classif" && hasLearnerProperties(learner, "prob"))
+    learner = setPredictType(learner, "prob")
+  mod = train(learner, task)
+  pred.train = predict(mod, task)
+  yhat = pred.train$data$response
+  perf.train = performance(pred.train, task = task, measures = measures)
+  if (cv > 0L) {
+    cv = crossval(learner, task, iters = 10L, measures = measures, show.info = FALSE)
+    perf.cv = cv$aggr
+    pred.cv = cv$pred
+  } else {
+    perf.cv = NA_real_
+  }
+
+  # 2d, 3d stuff
+  if (taskdim > 1L) {
+    x2n = features[2L]
+    x2 = data[, x2n]
+    if (taskdim == 3L) {
+      x3n = features[3L]
+      x3 = data[, x3n]
+    }
+  }
+
+  # grid for predictions
+  if (taskdim == 1L) {
+    grid = data.frame(x = seq(min(x1), max(x1), length.out = gridsize))
+  } else if (taskdim == 2L) {
+    # setup data frames for ggplot. grid = background, data = points
+    grid = expand.grid(
+      seq(min(x1), max(x1), length.out = gridsize),
+      seq(min(x2), max(x2), length.out = gridsize)
+    )
+  } else if (taskdim == 3L) {
+    grid = expand.grid(
+      seq(min(x1), max(x1), length.out = gridsize / 5),
+      seq(min(x2), max(x2), length.out = gridsize / 5),
+      seq(min(x3), max(x3), length.out = gridsize / 5)
+    )
+  }
+  colnames(grid) = features
+  pred.grid = predict(mod, newdata = grid)
+  grid[, target] = pred.grid$data$response
+
+  # set title
+  if (pretty.names) {
+    lrn.str = learner$short.name
+  } else {
+    lrn.str = learner$id
+  }
+  title = sprintf("%s: %s", lrn.str, paramValueToString(learner$par.set, learner$par.vals))
+  title = sprintf("%s\nTrain: %s; CV: %s", title, perfsToString(perf.train), perfsToString(perf.cv))
+
+  if (td$type == "classif") {
+    data$.err = if (err.mark == "train")
```

Comment:
I would change this to set `data$.err` in every branch of the `if`. That makes it easier to read.